tests/point_test.cpp

```diff
@@ -27,6 +27,78 @@ TEST_CASE( "rectangle_containment_raw", "[point]" )
     CHECK( !r2.contains( point( 0, 3 ) ) );
 }
 
+TEST_CASE( "rectangle_overlapping_inclusive", "[point]" )
+{
+    inclusive_rectangle<point> r1( point( 0, 0 ), point( 2,
+                                   2 ) ); // NOLINT(cata-use-named-point-constants)
+    inclusive_rectangle<point> r2( point( 2, 2 ), point( 3, 3 ) );
+    inclusive_rectangle<point> r3( point( 0, 0 ), point( 2,
+                                   1 ) ); // NOLINT(cata-use-named-point-constants)
+    inclusive_rectangle<point> r4( point( -2, -4 ), point( 4, -1 ) );
+    inclusive_rectangle<point> r5( point( -1, -3 ), point( 0, -2 ) );
+
+    CHECK( r1.overlaps( r1 ) );
+    CHECK( r1.overlaps( r2 ) );
+    CHECK( r1.overlaps( r3 ) );
+    CHECK( !r1.overlaps( r4 ) );
+    CHECK( !r1.overlaps( r5 ) );
+
+    CHECK( r2.overlaps( r1 ) );
+    CHECK( r2.overlaps( r2 ) );
+    CHECK( !r2.overlaps( r3 ) );
+    CHECK( !r2.overlaps( r4 ) );
+    CHECK( !r2.overlaps( r5 ) );
+
+    CHECK( r3.overlaps( r1 ) );
+    CHECK( !r3.overlaps( r2 ) );
+    CHECK( r3.overlaps( r3 ) );
+    CHECK( !r3.overlaps( r4 ) );
+    CHECK( !r3.overlaps( r5 ) );
+
+    CHECK( !r4.overlaps( r1 ) );
+    CHECK( !r4.overlaps( r2 ) );
+    CHECK( !r4.overlaps( r3 ) );
+    CHECK( r4.overlaps( r4 ) );
+    CHECK( r4.overlaps( r5 ) );
+    CHECK( r5.overlaps( r4 ) );
+}
+
+TEST_CASE( "rectangle_overlapping_half_open", "[point]" )
+{
+    half_open_rectangle<point> r1( point( 0, 0 ), point( 2,
+                                   2 ) ); // NOLINT(cata-use-named-point-constants)
+    half_open_rectangle<point> r2( point( 2, 2 ), point( 3, 3 ) );
+    half_open_rectangle<point> r3( point( 0, 0 ), point( 2,
+                                   1 ) ); // NOLINT(cata-use-named-point-constants)
```

Comment:
I feel like these are cases where you could happily use `point_zero`.  But if you really don't want to, then I think these `NOLINT` comments won't work when on the next line like this (maybe they do; looks like the error hasn't occurred yet).